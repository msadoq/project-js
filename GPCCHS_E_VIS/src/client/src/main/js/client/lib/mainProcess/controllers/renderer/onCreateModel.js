import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { add } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../documentManager';
import { getRootDir } from '../../../common/fmd';
import { getViewModule } from '../../../viewManager';

const root = getRootDir();

export default function ({ viewId }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), { viewId });

  const folder = absolutePath ? dirname(absolutePath) : root;
  const modelConfiguration = getViewModule(type).prepareConfigurationForModel(configuration);
  return getPathByFilePicker(folder, 'model', 'save', (err, path) => {
    saveViewAs(modelConfiguration, type, path, (errSaving) => {
      if (errSaving) {
        dispatch(add(viewId, 'danger', `Model unsaved ${errSaving}`));
      } else {
        dispatch(add(viewId, 'success', 'Model saved'));
      }
    });
  });
}
