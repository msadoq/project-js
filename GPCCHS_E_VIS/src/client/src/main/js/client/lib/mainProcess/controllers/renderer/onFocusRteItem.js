import asyncParallel from 'async/parallel';
import { getStore } from 'mainProcess/store';
import _reduce from 'lodash/reduce';
import _set from 'lodash/set';
import _indexOf from 'lodash/indexOf';
import {
  setRteCatalogs,
  setRteDomains,
  setRteItemNames,
  setRteFocusedItem,
} from 'store/actions/rte';
import { getRtd } from 'rtdManager';
import prepareDataToTree from 'rtdManager/prepareDataToTree';
import { add } from 'store/actions/messages';
import {
  getRteFocusedInfo,
} from 'store/reducers/rte';
import asyncMap from 'async/map';

export default function ({ sessionId, domainId, catalog, version, namespace, name, key }) {
  const { getState, dispatch } = getStore();

  const state = getState();
  const focusedInfo = getRteFocusedInfo(state);

  const rtd = getRtd();
  const requests = [];

  if (sessionId !== focusedInfo.session) {
    // focus domain, request catalogs
    requests.push((callback) => {
      rtd.getDatabase().getDomainListBySession(sessionId, (err, domains) => {
        if (err) {
          callback(err);
          return;
        }
        dispatch(setRteDomains(sessionId, domains));
        callback(null);
      });
    });
  }
  if (domainId !== focusedInfo.domain) {
    // focus session, request domains
    requests.push((callback) => {
      rtd.getDatabase().getCatalogAndVersionListByDomain(
        sessionId, domainId, (err, catalogList) => {
          if (err) {
            callback(err);
            return;
          }
          const catalogs = prepareDataToTree(catalogList);
          const catalogIdx = _indexOf(Object.keys(catalogList), catalog);
          _set(catalogs, ['children', catalogIdx, 'toggled'], true);
          _set(catalogs, ['children', catalogIdx, 'active'], true);
          dispatch(setRteCatalogs(sessionId, domainId, catalogs));
          callback(null);
        }
      );
    });
  }
  if (catalog !== focusedInfo.catalog || version !== focusedInfo.version ||
      namespace !== focusedInfo.namespace || name !== focusedInfo.name) {
    // focus catalog, request items
    requests.push((callback) => {
      rtd.getDatabase().getNamespaceList(catalog, version, (err, namespaceList) => {
        if (err) {
          callback(err);
          return;
        }
        asyncMap(namespaceList, (ns, done) => {
          rtd.getDatabase().getItemList(catalog, version, ns, (iErr, itemList) => {
            if (iErr) {
              done(iErr);
              return;
            }
            done(null, { ns, itemList });
          });
        }, (aErr, res) => {
          if (aErr) {
            callback(aErr);
            return;
          }
          const itemMap = _reduce(res, (previous, value) => {
            if (!value) {
              return previous;
            }
            const item = Object.assign({}, previous, {
              [value.ns]: value.itemList,
            });
            return item;
          }, {});
          const items = prepareDataToTree(itemMap);
          const namespaceIdx = _indexOf(Object.keys(itemMap), namespace);
          const nameIdx = _indexOf(itemMap[namespace], name);
          _set(items, ['children', namespaceIdx, 'toggled'], true);
          _set(items, ['children', namespaceIdx, 'children', nameIdx, 'active'], true);
          dispatch(setRteItemNames(catalog, version, items));
          callback(null);
        });
      });
    });
  }

  asyncParallel(requests, (err) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }
    dispatch(setRteFocusedItem(key, sessionId, domainId, catalog, version, namespace, name));
  });
}
