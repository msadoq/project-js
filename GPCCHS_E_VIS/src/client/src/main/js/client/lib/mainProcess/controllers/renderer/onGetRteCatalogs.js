// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : update catalog explorer controllers with new rtd connection
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// END-HISTORY
// ====================================================================

import { getStore } from '../../store';
import { getRtd } from '../../../rtdManager';
import prepareDataToTree from '../../../rtdManager/prepareDataToTree';
import { add } from '../../../store/actions/messages';
import {
  setRteCatalogs,
} from '../../../store/actions/rte';


export default function ({ sessionId, domainId }) {
  const { dispatch } = getStore();

  const rtd = getRtd();
  rtd.getDatabase().getCatalogAndVersionListByDomain(sessionId, domainId, (err, catalogList) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }
    const catalogs = prepareDataToTree(catalogList);
    dispatch(setRteCatalogs(sessionId, domainId, catalogs));
  });
}
