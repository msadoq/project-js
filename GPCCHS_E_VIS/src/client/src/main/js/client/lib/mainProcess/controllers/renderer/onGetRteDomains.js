import { getStore } from 'mainProcess/store';
import { add } from 'store/actions/messages';
import { getRtd } from 'rtdManager';
import {
  setRteDomains,
} from 'store/actions/rte';


export default function ({ sessionId }) {
  const { dispatch } = getStore();

  const rtd = getRtd();
  rtd.getDatabase().getDomainListBySession(sessionId, (err, domains) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }
    dispatch(setRteDomains(sessionId, domains));
  });
}
