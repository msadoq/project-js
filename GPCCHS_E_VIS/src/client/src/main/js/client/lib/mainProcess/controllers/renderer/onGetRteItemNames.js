// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : update catalog explorer controllers with new rtd connection
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// END-HISTORY
// ====================================================================

import asyncMap from 'async/map';
import _reduce from 'lodash/reduce';
import { getStore } from '../../store';
import { getRtd } from '../../../rtdManager';
import prepareDataToTree from '../../../rtdManager/prepareDataToTree';
import { add } from '../../../store/actions/messages';
import {
  setRteItemNames,
} from '../../../store/actions/rte';


export default function ({ catalog, version }) {
  const { dispatch } = getStore();

  const rtd = getRtd();
  rtd.getDatabase().getNamespaceList(catalog, version, (err, namespaceList) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }

    asyncMap(namespaceList, (namespace, done) => {
      rtd.getDatabase().getItemList(catalog, version, namespace, (iErr, itemList) => {
        if (iErr) {
          done(iErr);
          return;
        }
        done(null, { namespace, itemList });
      });
    }, (aErr, res) => {
      if (aErr) {
        dispatch(add('global', 'warning', err));
        return;
      }
      const itemMap = _reduce(res, (previous, value) => {
        if (!value) {
          return previous;
        }
        const item = Object.assign({}, previous, {
          [value.namespace]: value.itemList,
        });
        return item;
      }, {});
      const items = prepareDataToTree(itemMap);
      dispatch(setRteItemNames(catalog, version, items));
    });
  });
}
