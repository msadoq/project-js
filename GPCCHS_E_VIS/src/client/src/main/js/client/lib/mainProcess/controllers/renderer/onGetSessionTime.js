import getLogger from '../../../common/logManager';
import reply from '../../../common/ipc/reply';
import { server } from '../../ipc';

const logger = getLogger('main:controllers:renderer:onGetSessionTime');

export default function (queryId, sessionId) {
  server.requestSessionTime(sessionId, ({ err, timestamp }) => {
    if (err) {
      logger.error(err);
      reply(queryId);
      return;
    }
    reply(queryId, { timestamp });
  });
}
