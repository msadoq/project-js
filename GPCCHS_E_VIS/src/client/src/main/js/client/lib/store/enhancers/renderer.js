/**
 * Renderer process store enhancer
 *
 * - intercepts dispatched actions => forwards to main process
 * - receives state diff from main process => apply
 */

export default function rendererStoreEnhancer(previousStoreCreator) {
  if (process.type !== 'renderer') {
    throw new Error('rendererStoreEnhancer used in a non-renderer process');
  }

  return function createStore(reducer, initialState) {
    const store = previousStoreCreator(reducer, initialState);

    // TODO : enhance
    const storeDotDispatch = store.dispatch;
    store.dispatch = (action) => {
      console.log('VIMA ACTION on rendererStoreEnhancer');
      storeDotDispatch(action);
    };

    return store;
  };
}
