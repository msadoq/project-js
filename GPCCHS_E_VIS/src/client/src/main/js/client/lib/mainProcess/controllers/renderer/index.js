// ====================================================================
// HISTORY
// VERSION : 1.1.0 : : : 28/02/2017 : Initial version
// VERSION : 1.1.2 : DM : #3622 : 07/03/2017 : first draft on inspector: retrieve data from rtd on right-click
// VERSION : 1.1.2 : DM : #5822 : 16/03/2017 : resolve a rtd link in the inspector
// VERSION : 1.1.2 : DM : #5828 : 30/03/2017 : Add a F1 button in VIMA to open the docu wiki helper
// VERSION : 1.1.2 : DM : #5822 : 03/04/2017 : merge dev in working branch
// VERSION : 1.1.2 : DM : #5828 : 23/05/2017 : Move common/callback and common/ipc modules in client sub-component
// VERSION : 1.1.2 : DM : #6785 : 12/06/2017 : activate links in views .
// VERSION : 1.1.2 : DM : #5828 : 13/06/2017 : Move common/constants/ in client/ folder
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Cleanup client/ file organization and test helper modules
// VERSION : 1.1.2 : FA : ISIS-FT-2132 : 15/06/2017 : Ask to save before closing view or page
// VERSION : 1.1.2 : DM : #6700 : 16/06/2017 : Add store enhancers helpers code coverage and merge with dev
// VERSION : 1.1.2 : DM : #6700 : 26/06/2017 : Change windowProcess health management to dispatch a Redux action directly.
// VERSION : 1.1.2 : DM : #6785 : 29/06/2017 : Fix opening view link in a new page and read only path for link definition
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Remove unused onSavePage ipc renderer to mainProcess controllers
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Replace ipc openPage by askOpenPage redux action
// VERSION : 1.1.2 : FA : #7235 : 18/07/2017 : Clean old code in menuManager/IPC controller
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 18/07/2017 : Fix common/fmd, disable Links feature for now
// VERSION : 1.1.2 : DM : #6700 : 20/07/2017 : Remove obsolete onServerDebug interface, ipcs and model
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about openInspector .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about FMD_GET/FMD_CREATE and RELOAD_SESSIONS/GET_SESSION_TIME
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about documents .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Remove serverDebug ipc . .
// VERSION : 1.1.2 : DM : #6700 : 03/08/2017 : Merge branch 'dev' into dbrugne-data
// VERSION : 1.1.2 : FA : #7145 : 04/08/2017 : Clean IPC about opening wiki helper + create a store folder in mainProcess
// END-HISTORY
// ====================================================================

import constants from 'constants';

import handle from 'common/ipc/handle';
import reply from 'common/ipc/reply';

import onReduxCurrentState from './onReduxCurrentState';
import onReduxDispatch from './onReduxDispatch';
import onResolveLink from './onResolveLink';
import onResolveRteLink from './onResolveRteLink';
import onGetRteCatalogs from './onGetRteCatalogs';
import onGetRteItemNames from './onGetRteItemNames';
import onOpenRteItem from './onOpenRteItem';
import onFocusRteItem from './onFocusRteItem';
import onGetRteDomains from './onGetRteDomains';

const controller = {
  [constants.IPC_METHOD_REDUX_CURRENT_STATE]: (...args) => onReduxCurrentState(reply, ...args),
  [constants.IPC_METHOD_REDUX_DISPATCH]: onReduxDispatch,
  [constants.IPC_METHOD_RESOLVE_LINK]: onResolveLink,
  [constants.IPC_METHOD_GET_RTE_DOMAINS]: onGetRteDomains,
  [constants.IPC_METHOD_GET_RTE_CATALOGS]: onGetRteCatalogs,
  [constants.IPC_METHOD_GET_RTE_ITEM_NAMES]: onGetRteItemNames,
  [constants.IPC_METHOD_OPEN_RTE_ITEM]: onOpenRteItem,
  [constants.IPC_METHOD_FOCUS_RTE_ITEM]: onFocusRteItem,
  [constants.IPC_METHOD_RESOLVE_RTE_LINK]: onResolveRteLink,
};

export default (electronEvent, data) => handle(
  controller,
  data,
  response => electronEvent.sender.send('global', response)
);
