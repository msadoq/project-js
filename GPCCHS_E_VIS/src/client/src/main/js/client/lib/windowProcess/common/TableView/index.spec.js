import _ from 'lodash/fp';
import React from 'react';
import renderer from 'react-test-renderer';
import { Table } from '.';

const row1 = {
  type: 'row',
  data: { a: 'aaa', b: 'bbb', c: 'ccc' },
};

const row2 = {
  type: 'row',
  data: { a: 'AAA', b: 'BBB', c: 'CCC' },
};

const rows = [
  { ...row1, mainRow: row1 },
  { type: 'subrow_header_title', data: 'TRANSITIONS', mainRow: row1 },
  { type: 'subrow_header', mainRow: row1 },
  { type: 'subrow', mainRow: row1, data: { d: 'ddd', e: 'eee', f: 'fff' } },
  { type: 'subrow', mainRow: row1, data: { d: 'DDD', e: 'EEE', f: 'FFF' } },
  { ...row2, mainRow: row2 },
  { type: 'subrow_header_title', data: 'TRANSITIONS', mainRow: row2 },
  { type: 'subrow_header', mainRow: row2 },
  { type: 'subrow', mainRow: row2, data: { d: 'dddd', e: 'eeee', f: 'ffff' } },
  { type: 'subrow', mainRow: row2, data: { d: 'DDDD', e: 'EEEE', f: 'FFFF' } },
];

const propsStub = {
  search: {},
  onSearch: _.noop,
  enableSearch: true,
  onClickSearchIcon: _.noop,
  cols: ['a', 'b', 'c'],
  subCols: ['d', 'e', 'f'],
  sort: {
    column: 'a',
    mode: 'ASC',
  },
  toggleSort: _.noop,
  disableSelection: false,
  onMouseEnter: _.noop,
  onMouseLeave: _.noop,
  onCollapse: _.noop,
  onUncollapse: _.noop,
  onClickRow: _.noop,
  onScrollUp: _.noop,
  onScrollDown: _.noop,
  containerHeight: 450,
  rowHeight: 22,
  nbDisplayedRows: 10,
  rows,
};

test('TableView renders correctly', () => {
  const tree = renderer.create(
    <Table {...propsStub} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
