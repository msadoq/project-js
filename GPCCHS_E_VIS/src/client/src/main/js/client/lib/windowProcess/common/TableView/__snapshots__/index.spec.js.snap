// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView renders correctly 1`] = `
<table
  className="TableView container"
>
  <thead>
    <tr
      style={
        Object {
          "height": "22px",
        }
      }
    >
      <th
        className={undefined}
      >
        <span
          className="clickableMargin label label-default"
          onClick={[Function]}
          title="Disable search"
        >
          <span
            className="glyphicon glyphicon-search"
          />
        </span>
      </th>
      <th
        className="search"
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </th>
      <th
        className="search"
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </th>
      <th
        className="search"
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </th>
    </tr>
    <tr
      style={
        Object {
          "height": "22px",
        }
      }
    >
      <th
        className=""
        onClick={[Function]}
      >
        
      </th>
      <th
        className="header"
        onClick={[Function]}
      >
        <span
          className="sortArrow label label-default"
        >
          <span
            className="glyphicon glyphicon-chevron-up"
          />
        </span>
        a
      </th>
      <th
        className="header"
        onClick={[Function]}
      >
        b
      </th>
      <th
        className="header"
        onClick={[Function]}
      >
        c
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="tableRow row-typeof-row"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
          "width": "100%",
        }
      }
      title=""
    >
      <td
        className="collapseColumn"
      >
        <span
          className="clickableMargin label label-default"
          onClick={[Function]}
          title="Expand"
        >
          <span
            className="glyphicon glyphicon-plus"
          />
        </span>
      </td>
      <td
        className=""
      >
        aaa
      </td>
      <td
        className=""
      >
        bbb
      </td>
      <td
        className=""
      >
        ccc
      </td>
    </tr>
    <tr
      className="dummyRow"
    >
      o
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow_header_title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
        }
      }
      title=""
    >
      <th
        className="collapseColumn"
      />
      <th
        colSpan={3}
        style={
          Object {
            "fontStyle": "italic",
          }
        }
      >
        TRANSITIONS
      </th>
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow_header"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
        }
      }
      title=""
    >
      <th
        className="collapseColumn"
      >
        
      </th>
      <th
        className=""
      >
        d
      </th>
      <th
        className=""
      >
        e
      </th>
      <th
        className=""
      >
        f
      </th>
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
          "width": "100%",
        }
      }
      title=""
    >
      <td
        className="collapseColumn"
      />
      <td
        className=""
      >
        ddd
      </td>
      <td
        className=""
      >
        eee
      </td>
      <td
        className=""
      >
        fff
      </td>
    </tr>
    <tr
      className="tableSubRow lastSubrowFromRange row-typeof-subrow"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
          "width": "100%",
        }
      }
      title=""
    >
      <td
        className="collapseColumn"
      />
      <td
        className=""
      >
        DDD
      </td>
      <td
        className=""
      >
        EEE
      </td>
      <td
        className=""
      >
        FFF
      </td>
    </tr>
    <tr
      className="dummyRow"
    >
      o
    </tr>
    <tr
      className="tableRow row-typeof-row"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
          "width": "100%",
        }
      }
      title=""
    >
      <td
        className="collapseColumn"
      >
        <span
          className="clickableMargin label label-default"
          onClick={[Function]}
          title="Expand"
        >
          <span
            className="glyphicon glyphicon-plus"
          />
        </span>
      </td>
      <td
        className=""
      >
        AAA
      </td>
      <td
        className=""
      >
        BBB
      </td>
      <td
        className=""
      >
        CCC
      </td>
    </tr>
    <tr
      className="dummyRow"
    >
      o
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow_header_title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
        }
      }
      title=""
    >
      <th
        className="collapseColumn"
      />
      <th
        colSpan={3}
        style={
          Object {
            "fontStyle": "italic",
          }
        }
      >
        TRANSITIONS
      </th>
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow_header"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
        }
      }
      title=""
    >
      <th
        className="collapseColumn"
      >
        
      </th>
      <th
        className=""
      >
        d
      </th>
      <th
        className=""
      >
        e
      </th>
      <th
        className=""
      >
        f
      </th>
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
          "width": "100%",
        }
      }
      title=""
    >
      <td
        className="collapseColumn"
      />
      <td
        className=""
      >
        dddd
      </td>
      <td
        className=""
      >
        eeee
      </td>
      <td
        className=""
      >
        ffff
      </td>
    </tr>
    <tr
      className="tableSubRow row-typeof-subrow"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "height": "22px",
          "width": "100%",
        }
      }
      title=""
    >
      <td
        className="collapseColumn"
      />
      <td
        className=""
      >
        DDDD
      </td>
      <td
        className=""
      >
        EEEE
      </td>
      <td
        className=""
      >
        FFFF
      </td>
    </tr>
  </tbody>
</table>
`;
